back-end/routes/users.js

```diff
@@ -40,18 +40,25 @@ router.post('/login', [
     const query = `SELECT password, id FROM users WHERE email = '${email}'`;
     const dbResults = await getRowFromDb(query);
 
-    // value changes if query worked.
-    const passwordCheck = dbResults.password;
-    const id = dbResults.id;
 
-    // check if user-supplied password matches the hashed password from the db
-    if(await bcrypt.compare(password, passwordCheck)){
-      // if it does, generate and return a jwt
-      const token = createJWT(id);
+    // update: need to check if dbResults returned undefined (i.e. does user exist?)
+    if(typeof dbResults !== 'undefined') {
```

Comment:
This is a nit, but I prefer having code written in the following way:

```
if (databaseQueryWentWrong) {
   // handle the error
   // respond with 400 res
}

// Do stuff where things work
```

That way, you can get the error case out of the way and it doesn't make the happy path enclosed in a conditional.